core/src/main/java/org/frankframework/jdbc/JdbcTransactionalStorage.java

```diff
@@ -566,122 +542,64 @@ protected String storeMessageInDatabase(Connection conn, String messageId, Strin
 				stmt.setTimestamp(++parPos, null);
 			}
 
-			if (!isStoreFullMessage()) {
-				if (isOnlyStoreWhenMessageIdUnique()) {
-					JdbcUtil.setParameter(stmt, ++parPos, messageId, getDbmsSupport().isParameterTypeMatchRequired());
-					stmt.setString(++parPos, getSlotId());
-				}
-				stmt.execute();
-				return null;
-			}
-			if (!dbmsSupport.mustInsertEmptyBlobBeforeData()) {
+			if (isStoreFullMessage()) {
 				int blobColumnIndex = ++parPos;
 				Object blobHandle = dbmsSupport.getBlobHandle(stmt, blobColumnIndex);
 				try (ObjectOutputStream oos = new ObjectOutputStream(getBlobOutputStream(dbmsSupport, blobHandle, stmt, blobColumnIndex, isBlobsCompressed()))) {
 					oos.writeObject(message);
 				}
 				dbmsSupport.applyBlobParameter(stmt, blobColumnIndex, blobHandle);
-
-				if (isOnlyStoreWhenMessageIdUnique()) {
-					stmt.setString(++parPos, messageId);
-					stmt.setString(++parPos, getSlotId());
-				}
-				stmt.execute();
-				try (ResultSet rs = stmt.getGeneratedKeys()) {
-					if (rs.next() && rs.getString(1) != null) {
-						return "<id>" + rs.getString(1) + "</id>";
-					}
-				}
-
-				boolean isMessageDifferent = isMessageDifferent(conn, messageId, message);
-				String resultString = createResultString(isMessageDifferent);
-				log.warn("MessageID [{}] already exists", messageId);
-				if (isMessageDifferent) {
-					log.warn("Message with MessageID [{}] is not equal", messageId);
-				}
-				return resultString;
-			}
-			if (isOnlyStoreWhenMessageIdUnique()) {
-				stmt.setString(++parPos, messageId);
-				stmt.setString(++parPos, getSlotId());
 			}
+
 			stmt.execute();
-			updateCount = stmt.getUpdateCount();
-			if (log.isDebugEnabled()) {
-				log.debug("update count for insert statement: " + updateCount);
-			}
-		}
-		if (updateCount > 0) {
-			if (log.isDebugEnabled()) {
-				log.debug("preparing select statement [{}]", selectKeyQuery);
-			}
-			// retrieve the key
-			String newKey;
-			try (PreparedStatement stmt = conn.prepareStatement(selectKeyQuery); ResultSet rs = stmt.executeQuery()) {
-				if (!rs.next()) {
-					throw new SenderException("could not retrieve key of stored message");
+			try (ResultSet rs = stmt.getGeneratedKeys()) {
+				if (rs.next() && rs.getString(1) != null) {
+					return "<id>" + rs.getString(1) + "</id>";
 				}
-				newKey = rs.getString(1);
 			}
+			log.warn("No keys generated after INSERT statement");
 
-			// and update the blob
-			if (log.isDebugEnabled()) {
-				log.debug("preparing update statement [{}]", updateBlobQuery);
-			}
-			try (PreparedStatement stmt = conn.prepareStatement(updateBlobQuery);){
-				stmt.clearParameters();
-				stmt.setString(1, newKey);
+			return null;
```

Comment:
With previous queries this was valid and indicated that the message had already been inserted, however now this should not happen.
I believe.

Perhaps this should be made an error actually.